src/test/java/com/puppycrawl/tools/checkstyle/JavaAstVisitorTest.java

```diff
@@ -190,4 +204,34 @@ public void testNullSelfInAddLastSibling() throws Exception {
                 .that(addLastSibling.invoke(JavaAstVisitor.class, null, null))
                 .isNull();
     }
+
+    @Test
+    public void testNoStackOverflowOnDeepStringConcat() throws Exception {
```

Comment:
Moved this test here since clearing the DFA in conjunction with the hack we use to limit resources was causing intermittent StackOverflow errors. This test is really for `JavaAstVisitor` anyway, so it makes more sense to have it here.